app/uk/gov/hmrc/tradergoodsprofilesrouter/models/response/eis/payloads/PatchRecordPayload.scala

```diff
@@ -58,4 +58,21 @@ object UpdateRecordPayload {
       comcodeEffectiveFromDate = request.comcodeEffectiveFromDate.map(_.truncatedTo(ChronoUnit.SECONDS)),
       comcodeEffectiveToDate = request.comcodeEffectiveToDate.map(_.truncatedTo(ChronoUnit.SECONDS))
     )
+
+  def apply(eori: String, recordId: String, request: UpdateRecordRequest): PatchRecordPayload =
```

Comment:
I think it would be simpler to split this into another case class, I get why you have done this, but for readability going forward I think it would be easier to have an `UpdateRecordRequest` and a `PatchRecordRequest`